talk/owt/sdk/p2p/p2ppeerconnectionchannel.cc

```diff
@@ -692,7 +695,6 @@ void P2PPeerConnectionChannel::OnIceConnectionChange(
       }).detach();
       break;
     case webrtc::PeerConnectionInterface::kIceConnectionClosed:
-      ended_ = true;
```

Comment:
is this state only hit when we trigger pc->Close() or can it be triggered o/w? It should be terminal

Reply:
It is only triggered by pc->Close, which is irreversible, so it will only be triggered once. But, ended_ needs to be set inside a lock, which is why I moved it to ClosePeerConnection().